src/test/ui/async-await/issue-62658.rs

```diff
@@ -0,0 +1,29 @@
+// This test created a generator whose size was not rounded to a multiple of its
+// alignment. This caused an assertion error in codegen.
+
+// build-pass
+// edition:2018
+
+#![feature(async_await)]
+
+async fn noop() {}
+
+async fn foo() {
+    // This suspend should be the largest variant.
+    {
+        let x = [0u8; 17];
+        noop().await;
+        println!("{:?}", x);
```

Comment:
Instead of `println` you could use either `drop` is define a `fn dummy<T>(_: T) {}` and use that